sdk/python/kfp/compiler/compiler.py

```diff
@@ -825,11 +848,10 @@ def _create_workflow(self,
     # will be resolved immediately in place when being to each component.
     default_param_values = OrderedDict()
 
-    if getattr(pipeline_func, 'output_directory', None):
-      dsl_pipeline_root = dsl.PipelineParam(
-          name=dsl.ROOT_PARAMETER_NAME, value=pipeline_func.output_directory)
-      pipeline_func.output_directory = dsl_pipeline_root
-      params_list.append(dsl_pipeline_root)
+    if self._pipeline_root:
+      params_list.append(self._pipeline_root)
+    if self._pipeline_name:
+      params_list.append(self._pipeline_name)
```

Comment:
Adding `pipeline_root` and `pipeline_name` into the `params_list` implies that the name `pipeline-name` and `pipeline-output-directory` now becomes the reserved parameter names that user cannot use the same names for pipeline function arguments.

So before adding them to the list, we should check if they already exist, and throw errors if they do.

Feel free to leave a `TODO(chensun): ...` here, and I can address this later.

Reply:
Added a check. PTAL.

Reply:
On a second thought, I realize `pipeline-name` and `pipeline-output-directory` are not legal Python identifiers while our pipeline definition must be a Python function. So I now think there won't be any conflict and we don't need the check at all? Sorry about the original misleading comment.

Reply:
Done.